### PR TITLE
Added seperate buttons for Start / Stop on Recording and Streaming - v2

### PR DESCRIPTION
**Pull request by mattehalen**
lets see if this one works better

### CODE REVIEW

**Comment by eric** [comment]
Looks great!

**Review by eric — APPROVED** [review]
This looks great and I could see using this.

**Comment by mattehalen** [comment]
made some more updates here if someone wanna take a look and approve.

File: index.js [hunk 1]
```diff
@@ -359,6 +359,42 @@ instance.prototype.actions = function() {
 	}
 
 	self.setActions({
+		'enable_StudioMode': {
```
Comment (eric) [inline index.js:362]:
```suggestion
		'enable_studio_mode': {
```

File: index.js [hunk 2]
```diff
@@ -359,6 +359,42 @@ instance.prototype.actions = function() {
...
+			label: 'Enable StudioMode',
+		},
+		'disable_StudioMode': {
```
Comment (eric) [inline index.js:365]:
```suggestion
		'disable_studio_mode': {
```

File: index.js [hunk 3]
```diff
@@ -359,6 +359,42 @@ instance.prototype.actions = function() {
...
+			label: 'Disable StudioMode',
+		},
+		'toggle_StudioMode': {
```
Comment (eric) [inline index.js:368]:
```suggestion
		'toggle_studio_mode': {
```

File: index.js [hunk 4]
```diff
@@ -359,6 +359,42 @@ instance.prototype.actions = function() {
...
+		'resume_recording': {
+			label: 'Resume Recording',
+		},
+		'start_streaming': {
+			label: 'Start Streaming',
+		},
+		'stop_streaming': {
+			label: 'Stop Streaming',
+		},
+		'start_ReplayBuffer': {
```
Comment (eric) [inline index.js:389]:
```suggestion
		'start_replay_buffer': {
```

File: index.js [hunk 5]
```diff
@@ -359,6 +359,42 @@ instance.prototype.actions = function() {
...
+			label: 'Start ReplayBuffer  ',
+		},
+		'stop_ReplayBuffer': {
```
Comment (eric) [inline index.js:392]:
```suggestion
		'stop_replay_buffer': {
```

File: index.js [hunk 6]
```diff
@@ -359,6 +359,42 @@ instance.prototype.actions = function() {
...
+			label: 'Stop ReplayBuffer  ',
+		},
+		'save_ReplayBuffer': {
```
Comment (eric) [inline index.js:395]:
```suggestion
		'save_replay_buffer': {
```

File: index.js [hunk 7]
```diff
@@ -545,6 +581,42 @@ instance.prototype.action = function(action) {
 	}
 
 	switch (action.action) {
+		case 'enable_StudioMode':
```
Comment (eric) [inline index.js:584]:
```suggestion
		case 'enable_studio_mode':
```

File: index.js [hunk 8]
```diff
@@ -545,6 +581,42 @@ instance.prototype.action = function(action) {
...
+			handle = self.obs.send('EnableStudioMode');
+			break;
+		case 'disable_StudioMode':
```
Comment (eric) [inline index.js:587]:
```suggestion
		case 'disable_studio_mode':
```

File: index.js [hunk 9]
```diff
@@ -545,6 +581,42 @@ instance.prototype.action = function(action) {
...
+			handle = self.obs.send('DisableStudioMode');
+			break;
+		case 'toggle_StudioMode':
```
Comment (eric) [inline index.js:590]:
```suggestion
		case 'toggle_studio_mode':
```

File: index.js [hunk 10]
```diff
@@ -545,6 +581,42 @@ instance.prototype.action = function(action) {
...
+		case 'resume_recording':
+			handle = self.obs.send('ResumeRecording');
+			break;
+		case 'start_streaming':
+			handle = self.obs.send('StartStreaming');
+			break;
+		case 'stop_streaming':
+			handle = self.obs.send('StopStreaming');
+			break;
+		case 'start_ReplayBuffer':
```
Comment (eric) [inline index.js:611]:
```suggestion
		case 'start_replay_buffer':
```

File: index.js [hunk 11]
```diff
@@ -545,6 +581,42 @@ instance.prototype.action = function(action) {
...
+			handle = self.obs.send('StartReplayBuffer');
+			break;
+		case 'stop_ReplayBuffer':
```
Comment (eric) [inline index.js:614]:
```suggestion
		case 'stop_replay_buffer':
```

File: index.js [hunk 12]
```diff
@@ -545,6 +581,42 @@ instance.prototype.action = function(action) {
...
+			handle = self.obs.send('StopReplayBuffer');
+			break;
+		case 'save_ReplayBuffer':
```
Comment (eric) [inline index.js:617]:
```suggestion
		case 'save_replay_buffer':
```

**Review by eric — CHANGES_REQUESTED** [review]
There are some issues with naming mixing snake case and camel case.

**Comment by mattehalen** [comment]
> There are some issues with naming mixing snake case and camel case.

These should be fixed now

**Pull request merged**